lib/application_insights/rack/track_request.rb

```diff
@@ -82,6 +90,33 @@ def format_request_duration(duration_seconds)
 
         Time.at(duration_seconds).gmtime.strftime("00.%H:%M:%S.%7N")
       end
+
+      def request_id_header(env)
+        request_id_header = env['HTTP_REQUEST_ID']
+        valid_request_id_header = request_id_header && request_id_header[0] == '|'
+
+        length = valid_request_id_header ? 8 : 16
+        id = SecureRandom.hex(length / 2)
+
+        if valid_request_id_header
+          request_id_has_end = %w[. _].include?(request_id_header[-1])
+          request_id_header << '.' unless request_id_has_end
+
+          return "#{request_id_header}#{id}."
+        end
+
+        "|#{id}."
+      end
+
+      def operation_id(id)
+        # Returns the root ID from the '|' to the first '.' if any.
+        root_end = id.index('.')
```

Comment:
the id could end with '_'

Reply:
I will update this and have opened a PR to update the azure docs here:

https://github.com/MicrosoftDocs/azure-docs/pull/7837